lightning/src/ln/chan_utils.rs

```diff
@@ -616,12 +629,17 @@ pub fn build_htlc_transaction(commitment_txid: &Txid, feerate_per_kw: u32, conte
 	} else {
 		htlc_success_tx_weight(opt_anchors)
 	};
-	let total_fee = feerate_per_kw as u64 * weight / 1000;
+	let output_value = if opt_anchors {
+		htlc.amount_msat / 1000
+	} else {
+		let total_fee = feerate_per_kw as u64 * weight / 1000;
+		htlc.amount_msat / 1000 - total_fee
+	};
 
 	let mut txouts: Vec<TxOut> = Vec::new();
 	txouts.push(TxOut {
 		script_pubkey: get_revokeable_redeemscript(revocation_key, contest_delay, broadcaster_delayed_payment_key).to_v0_p2wsh(),
-		value: htlc.amount_msat / 1000 - total_fee //TODO: BOLT 3 does not specify if we should add amount_msat before dividing or if we should divide by 1000 before subtracting (as we do here)
```

Comment:
Do we need to keep this TODO for the above change?

As far as I can infer from the spec it seems that what we do here isn't wrong but probably should be made clearer there. At least "`Multiply feerate_per_kw by xxx (xxx if option_anchor_outputs applies) and divide by 1000 (rounding down)`" seems to match what we do.

Reply:
Ok, it's here: https://github.com/lightning/bolts/pull/1020